generateLJFF.py

```diff
@@ -3,39 +3,55 @@
 import sys
 
 import __main__ as main
-import numpy as np
 
-import ppafm as PPU
-import ppafm.fieldFFT as fFFT
-import ppafm.GridUtils as GU
-import ppafm.HighLevel as PPH
-from ppafm import basUtils, elements
+import ppafm
+from ppafm.HighLevel import computeLJ
 
-if __name__=="__main__":
-    HELP_MSG="""Use this program in the following way:
+if __name__ == "__main__":
+    HELP_MSG = """Use this program in the following way:
     %s -i <filename>
 
     Supported file fromats are:
        * xyz
-    """ %os.path.basename(main.__file__)
+    """ % os.path.basename(
+        main.__file__
+    )
     from optparse import OptionParser
+
     parser = OptionParser()
-    parser.add_option("-i", "--input",      action="store", type="string",  help="Input file, supported formats are:\n.xyz\n.cube,.xsf")
-    parser.add_option("-f","--data_format", action="store" , type="string", help="Specify the output format of the vector and scalar field. Supported formats are: xsf,npy", default="xsf")
-    parser.add_option("--noPBC",            action="store_false",           help="pbc False", dest="PBC", default=None)
-    parser.add_option("-E", "--energy",     action="store_true",            help="Compue potential energ y(not just Force)", default=False)
-    parser.add_option("--ffModel",          action="store",                 help="kind of potential 'LJ','Morse','vdW' ", default='LJ')
```

Comment:
Even though these lines are really wide, I would argue that since it's so nicely aligned it reads better than automatically formatted version. You can see all of the arguments at a glance.

Reply:
The only problem is that we are far from being consistent throughout the whole code. Maybe here it reads nice, but in many other places, the number of spaces and alignment is random.

I think consistency is better than having one specific part of the code nicely aligned.

Reply:
I see a couple of places where I think it improves legibility, a couple of places like this where it makes it worse, but mostly just kind of neutral. I think it's quite situation dependent what kind of style is more legible.

Overall, I think these kind of aesthetic changes are not that important and rather the actually important things to make code readable are things like
- Divide the operations in the code into lines or blocks in a sensible way that make the flow of the logic clear.
- Write enough comments to make clear what each part of the code is doing.
- Make the overall structure of the code consistent and make a consistent API. I think switching to PEP8 naming scheme would be a positive change in this regard.

I don't see black helping much with this, but I also don't see value in debating very long about it, so maybe it's better if you just go ahead and implement it.

Reply:
> * Divide the operations in the code into lines or blocks in a sensible way that make the flow of the logic clear.

Black doesn't read the logic, it is a code formatter. We have many other pre-commit hooks that help with a better writing style.

> * Make the overall structure of the code consistent

Here black is definitely helpful.

> I think switching to PEP8 naming scheme would be a positive change in this regard.

I opened an issue about this a few days ago: https://github.com/Probe-Particle/ppafm/issues/93

Reply:
Yes, this is exactly example where I worry that black will destroy something which helps me read the code just for the sake of some arbitrary convention. I have really problem reading long code, and I need to organize the code into blocks/tables to help me see the connections (help see the common patterns). Even when I read other people codes I often have to reformat them into such blocks/tables in order to help me see the patterns and understand it.

The importain thing is that this block-structure I have to hand-craft, which is tedious, and Black would destroy it in one go, and there is no way back without doing all that tedious human-labor-intensive re-formating again. Therefore I'm against using black.

I would be quite happy to use some automatic formatar, which can do this for me - i.e. where I can set the rules to do this table-like or block-like alignement automatically.

But also I agree with @NikoOinonen  that I don't want to spend much time on this cosmetic issue (i.e. carefully set the rules for automatic formater).